File: INSERT Image into SQL-Server using stored procedures/[C#][VB]-INSERT Image into SQL-Server using stored procedures/C# and VB/Demo_vb/MainForm.vb
Public Class MainForm
    Private bs As New BindingSource
    Private ops As New DatabaseImageOperations
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles cmdInsertImage.Click
        If OpenFileDialog1.ShowDialog = DialogResult.OK Then
            Dim Identifier As Integer = 0
            Dim fileBytes As Byte() = IO.File.ReadAllBytes(OpenFileDialog1.FileName)
            If ops.InsertImage(fileBytes, Now.ToShortTimeString, Identifier) = Success.Okay Then
                CType(bs.DataSource, DataTable).Rows.Add(New Object() {Identifier, fileBytes, Now.ToShortTimeString})
            End If
        End If
    End Sub
    Private Sub MainForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        DataGridView1.AutoGenerateColumns = False
        bs.DataSource = ops.DataTable
        DataGridView1.DataSource = bs

        Dim ImageBinding As New Binding("Image", bs, "ImageData")
        AddHandler ImageBinding.Format, AddressOf BindImage
        PictureBox1.DataBindings.Add(ImageBinding)
    End Sub
    Private Sub BindImage(ByVal sender As Object, ByVal e As ConvertEventArgs)
        If e.DesiredType Is GetType(Image) Then
            Dim ms As New System.IO.MemoryStream(CType(e.Value, Byte()))
            Dim Logo As Image = Image.FromStream(ms)
            e.Value = Logo
        End If
    End Sub
End Class

File: INSERT Image into SQL-Server using stored procedures/[C#][VB]-INSERT Image into SQL-Server using stored procedures/C# and VB/WindowsApplication_cs/MainForm.cs
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using WindowsApplication_cs.Classes;

namespace WindowsApplication_cs
{
    public partial class MainForm : Form
    {
        BindingSource bs = new BindingSource();
        DatabaseImageOperations ops = new DatabaseImageOperations();
        public MainForm()
        {
            InitializeComponent();
        }
        void BindImage(object sender, ConvertEventArgs e)
        {
            if (e.DesiredType == typeof(Image))
            {
                System.IO.MemoryStream ms = new System.IO.MemoryStream((byte[])e.Value);
                Image image = Image.FromStream(ms);
                e.Value = image;
            }
        }
        void Form1_Load(object sender, EventArgs e)
        {
            DataGridView1.AutoGenerateColumns = false;
            bs.DataSource = ops.DataTable();
            DataGridView1.DataSource = bs;

            Binding ImageBinding = new Binding("Image", bs, "ImageData");
            ImageBinding.Format += BindImage;
            PictureBox1.DataBindings.Add(ImageBinding);
        }

        void cmdInsertImage_Click(object sender, EventArgs e)
        {
            SelectImageForm f = new SelectImageForm();
            try
            {
                if (f.ShowDialog() == DialogResult.OK)
                {
                    int Identifier = 0;
                    var fileBytes = System.IO.File.ReadAllBytes(f.FileName);
                    if (ops.InsertImage(fileBytes, f.Description, ref Identifier) == Success.Okay)
                    {
                        ((DataTable)bs.DataSource).Rows.Add(new object[] { Identifier, fileBytes, f.Description });
                        var index = bs.Find("ImageId", Identifier);
                        if (index > -1)
                        {
                            bs.Position = index;
                        }
                    }
                }
            }
            finally
            {
                f.Dispose();
            }
        }
    }
}
